libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/BasicAssertionsTest.kt

```diff
@@ -205,6 +205,28 @@ class BasicAssertionsTest {
     fun testExpectFails() {
         checkFailedAssertion { expect(1) { 2 } }
     }
+
+    @Test
+    fun testAssertIs() {
+        val s: Any = "test"
+        assertIs<String>(s)
+        assertEquals(4, s.length)
+    }
+    
+    @Test
+    fun testAssertIsFails() {
+        checkFailedAssertion { assertIs<Int>("test") }
```

Comment:
Could you inspect the assertion returned by `checkFailedAssertion` and check that its message contains the information we expect?